test/clj/rems/db/test_entitlements.clj

```diff
@@ -195,36 +197,56 @@
 
           (testing "removing a member ends entitlements"
             (entitlements-poller/run)
-            (is (= 5 (count (stub/recorded-requests server))))
+            (is (= 6 (count (stub/recorded-requests server))))
             (testing "db"
               (is (= [[uid "resource1"] [uid "resource2"]]
                      (map (juxt :userid :resid) (db/get-entitlements {:application app-id :is-active? true})))))
             (testing "POST"
-              (let [data (nth (stub/recorded-requests server) 4)
-                    target (:path data)
-                    body (cheshire/parse-string (get-in data [:body "postData"]) keyword)]
-                (is (= "/remove" target))
+              (let [data (take 2 (drop 4 (stub/recorded-requests server)))
```

Comment:
this test has grown pretty big. could we split it into separate tests or perhaps make some helpers for checking the entitlements & POSTs? also, is there a way to reset the stub to get rid of the magic `(take 2 (drop 4 ...))`?

Reply:
I refactored the implementation to use a helper so this tracking is not required anymore.